maestro-orchestra/src/main/java/maestro/orchestra/Orchestra.kt

```diff
@@ -114,39 +114,21 @@ class Orchestra(
         return true
     }
 
-    private fun executeCommand(command: MaestroCommand) {
-        when {
-            command.tapOnElement != null -> command.tapOnElement
-                ?.let {
-                    tapOnElement(
-                        it,
-                        it.retryIfNoChange ?: true,
-                        it.waitUntilVisible ?: true,
-                    )
-                }
-            command.tapOnPoint != null -> command.tapOnPoint
-                ?.let {
-                    tapOnPoint(
-                        it,
-                        it.retryIfNoChange ?: true,
-                    )
-                }
-            command.backPressCommand != null -> maestro.backPress()
-            command.scrollCommand != null -> maestro.scrollVertical()
-            command.swipeCommand != null -> command.swipeCommand
-                ?.let { swipeCommand(it) }
-            command.assertCommand != null -> command.assertCommand
-                ?.let { assertCommand(it) }
-            command.inputTextCommand != null -> command.inputTextCommand
-                ?.let { inputTextCommand(it) }
-            command.launchAppCommand != null -> command.launchAppCommand
-                ?.let { launchAppCommand(it) }
-            command.openLinkCommand != null -> command.openLinkCommand
-                ?.let { openLinkCommand(it) }
-            command.pressKeyCommand != null -> command.pressKeyCommand
-                ?.let { pressKeyCommand(it) }
-            command.eraseTextCommand != null -> command.eraseTextCommand
-                ?.let { eraseTextCommand(it) }
+    private fun executeCommand(command: Command?) {
+        when (command) {
```

Comment:
nit: let's do `return when (command) {}` to enforce `when` being an exhaustive list

Reply:
We could do that. I believe `ApplyConfigurationCommand` would be a no-op in that case?

Also, can we make the `command` parameter non-null? Are there situations where we need to deal with `null` values?

Reply:
> I believe ApplyConfigurationCommand would be a no-op in that case?

Exactly

> Also, can we make the command parameter non-null? Are there situations where we need to deal with null values?

I thought about that, but I think it makes sense to keep it nullable for backwards compatibility (i.e. if we encounter an unknown command in JSON)

Reply:
Got it, then I'll go ahead and make the change and keep the nullability as is.

Reply:
This change is done 👍🏽